airbyte-server/src/main/java/io/airbyte/server/handlers/helpers/CatalogConverter.java

```diff
@@ -66,7 +67,18 @@ private static io.airbyte.protocol.models.AirbyteStream toProtocol(final Airbyte
         }
       }
       // Only include the selected fields.
-      List<String> selectedFieldNames = config.getSelectedFields().stream().map((field) -> field.getFieldPath().get(0)).collect(Collectors.toList());
+      final Set<String> selectedFieldNames =
+          config.getSelectedFields().stream().map((field) -> field.getFieldPath().get(0)).collect(Collectors.toSet());
+      // TODO(mfsiega-airbyte): we only check the top level of the cursor/primary key fields because we
+      // don't support filtering nested fields yet.
+      if (!selectedFieldNames.contains(config.getCursorField().get(0))) {
+        throw new JsonValidationException("Cursor field cannot be de-selected");
```

Comment:
I'm trying to think if there could be a case where a non-cursor field shares a top-level key with a cursor field, and therefore can't be de-selected. Even if that is the case, probably better to merge this as-is for now because we're planning on adding support for nested fields relatively soon anyways

Reply:
Yeah as it is today we only allow selecting top-level fields (which implies all of the children) - so the only way _to_ deselect that non-cursor-field-which-shares-a-top-level-key-with-a-cursor is to deselect the top-level field, which in turn deselects the cursor...

But indeed we should support nested fields soon, at least on the backend.

Reply:
Cool, makes sense to me, thanks for confirming!